Mathlib/Algebra/Group/UniqueProds.lean

```diff
@@ -212,8 +239,44 @@ is 'very monotone', then `A` also has `UniqueSums`."]
 instance (priority := 100) Covariants.to_uniqueProds {A} [Mul A] [LinearOrder A]
     [CovariantClass A A (· * ·) (· ≤ ·)] [CovariantClass A A (Function.swap (· * ·)) (· < ·)]
     [ContravariantClass A A (· * ·) (· ≤ ·)] : UniqueProds A where
-      uniqueMul_of_nonempty {A} {B} hA hB :=
-        ⟨_, A.min'_mem ‹_›, _, B.min'_mem ‹_›, fun a b ha hb ab ↦
-        eq_and_eq_of_le_of_le_of_mul_le (Finset.min'_le _ _ ‹_›) (Finset.min'_le _ _ ‹_›) ab.le⟩
+  uniqueMul_of_nonempty {A B} hA hB :=
+    ⟨_, A.min'_mem ‹_›, _, B.min'_mem ‹_›, fun a b ha hb ab ↦
+      eq_and_eq_of_le_of_le_of_mul_le (Finset.min'_le _ _ ‹_›) (Finset.min'_le _ _ ‹_›) ab.le⟩
 #align covariants.to_unique_prods Covariants.to_uniqueProds
 #align covariants.to_unique_sums Covariants.to_uniqueSums
+
+namespace UniqueProds
+
+@[to_additive (attr := nontriviality, simp)]
+theorem of_subsingleton {G : Type*} [Mul G] [Subsingleton G] :
+    UniqueProds G :=
+  ⟨fun {A B} ⟨a, hA⟩ ⟨b, hB⟩ ↦ ⟨a, hA, b, hB, by simp⟩⟩
```

Comment:
maybe more readable as
```suggestion
theorem of_subsingleton {G : Type*} [Mul G] [Subsingleton G] : UniqueProds G where
  uniqueMul_of_nonempty {A B} | ⟨a, hA⟩ ⟨b, hB⟩ => ⟨a, hA, b, hB, by simp⟩
```
(untested)

Reply:
It turns out that you need a comma between the angle brackets.